src/main/java/teammates/storage/api/StudentsDb.java

```diff
@@ -407,45 +407,60 @@ public void updateStudent(String courseId, String email, String newName,
             String lastName = StringHelper.splitName(newName)[1];
             
             if (isEmailChanged) {
-                CourseStudent newCourseStudent = new CourseStudent(newEmail, newName, newGoogleId, newComments,
-                                                                   courseId, newTeamName, newSectionName);
-                newCourseStudent.setLastName(lastName);
-                newCourseStudent.setCreatedAt(courseStudent.getCreatedAt());
-                if (keepUpdateTimestamp) {
-                    newCourseStudent.setLastUpdate(courseStudent.getUpdatedAt());
-                }
-                
+                CourseStudent newCourseStudent = new CourseStudent(newEmail, newName, newGoogleId,
+                        newComments,
+                        courseId, newTeamName, newSectionName);
                 try {
-                    createStudent(new StudentAttributes(newCourseStudent), hasDocument);
+                    updateStudentChangeEmail(newCourseStudent, lastName, courseStudent, hasDocument,
+                            keepUpdateTimestamp, courseId, email);
                 } catch (EntityAlreadyExistsException e) {
                     StudentAttributes existingStudent = (StudentAttributes) e.existingEntity;
                     String error = ERROR_UPDATE_EMAIL_ALREADY_USED
                             + existingStudent.getName() + "/" + existingStudent.getEmail();
                     throw new InvalidParametersException(error);
                 }
```

Comment:
you will need to look for and transfer `@SuppressWarnings("PMD.PreserveStackTrace")` from `updateStudent` to your extracted method. This will suppress the error you got initially.